csvcubed/csvcubed/readers/v1_0/columnschema.py

```diff
@@ -0,0 +1,459 @@
+"""
+Models
+------
+
+config.json V1.0 column mapping models.
+"""
+import uritemplate
+
+from dataclasses import dataclass
+from typing import List, Union, Optional, TypeVar
+from pathlib import Path
+
+from csvcubed.models.cube import (
+    CatalogMetadata,
+    CompositeQbCodeList,
+    DuplicatedQbConcept,
+    ExistingQbAttributeLiteral,
+    ExistingQbCodeList,
+    NewQbAttributeLiteral,
+    NewQbCodeList,
+    NewQbCodeListInCsvW, QbCodeList,
+)
+
+from csvcubedmodels.dataclassbase import DataClassBase
+
+from csvcubed.inputs import pandas_input_to_columnar_optional_str
+from csvcubed.models.cube.qb.components import (
+    NewQbDimension,
+    ExistingQbDimension,
+    NewQbAttribute,
+    NewQbAttributeValue,
+    ExistingQbAttribute,
+    NewQbUnit,
+    ExistingQbUnit,
+    QbMultiUnits,
+    QbMultiMeasureDimension,
+    QbMultiMeasureObservationValue,
+    QbSingleMeasureObservationValue,
+    ExistingQbMeasure,
+    NewQbMeasure,
+    QbObservationValue,
+)
+from csvcubed.inputs import PandasDataTypes
+from csvcubed.utils.uri import (
+    csvw_column_name_safe,
+    looks_like_uri,
+)
+
+T = TypeVar("T", bound=object)
+
+
+@dataclass
+class SchemaBaseClass(DataClassBase):
+    ...
```

Comment:
If there's no implementation, can you ensure that it implements the `ABC` abstract base class so it isn't possible to instantiate instances of it.

Reply:
done.